.github/actions/upload_test_artifacts/action.yaml

```diff
@@ -10,26 +10,44 @@ inputs:
   test_artifacts_key:
     description: "Artifact key used to store on-host test artifacts."
     required: true
-  test_targets_json:
-    description: "The test targets as a json list."
+  test_targets_json_file:
+    description: "The path to test targets json file."
     required: true
 runs:
   using: "composite"
   steps:
-    - name: Archive Test Artifacts
+    - name: Copy Runnables to Artifacts
+      if: inputs.upload_on_device_test_artifacts == 'true'
+      env:
+        TEST_TARGETS_JSON_FILE: ${{ inputs.test_targets_json_file }}
+      run: |
+        set -eux
+        cd src/
+        mkdir -p "${GITHUB_WORKSPACE}/artifacts/"
+        for target in $(cat "${TEST_TARGETS_JSON_FILE}" | jq -r '.executables | .[]'); do
+          cp "${target}" "${GITHUB_WORKSPACE}/artifacts/"
+        done
+      shell: bash
+    - name: Archive Test Dependencies
       env:
-        TEST_TARGETS_JSON: ${{ inputs.test_targets_json }}
+        TEST_TARGETS_JSON_FILE: ${{ inputs.test_targets_json_file }}
       run: |
         set -x
-        mkdir ${GITHUB_WORKSPACE}/artifacts
         cd src/
 
+        FLAGS=""
+        if [[ "${{ matrix.platform }}" == "android" ]]; then
+          FLAGS="--archive-per-target --use-android-deps-path --root-dir ."
+        elif [[ "${{ matrix.platform }}" == *raspi* ]]; then
+          FLAGS="--archive-per-target"
+        fi
+
         time vpython3 -u ./cobalt/build/archive_test_artifacts.py \
```

Comment:
Probably just have to call `mkdir -p "${GITHUB_WORKSPACE}/artifacts/"` here as well (before `archive_test_artifacts.py`) to fix on host tests

Reply:
Done